tests/js/server/aql/aql-index-join.js

```diff
@@ -162,6 +182,113 @@ const IndexJoinTestSuite = function () {
       }
     },
 
+    testEveryOtherMatchBig: function () {
+      const A = fillCollection("A", singleAttributeGenerator(10000, "x", x => x));
+      A.ensureIndex({type: "persistent", fields: ["x"]});
+      const B = fillCollection("B", singleAttributeGenerator(5000, "x", x => 2 * x));
+      B.ensureIndex({type: "persistent", fields: ["x"]});
+
+      const result = runAndCheckQuery(`
+        FOR doc1 IN A
+          SORT doc1.x
+          FOR doc2 IN B
+              FILTER doc1.x == doc2.x
+              RETURN [doc1, doc2]
+      `);
+
+      assertEqual(result.length, 5000);
+      for (const [a, b] of result) {
+        assertEqual(a.x, b.x);
+      }
+    },
+
+    testEveryOtherMatchUniqueSmall: function () {
+      const A = fillCollection("A", singleAttributeGenerator(10, "x", x => x));
+      A.ensureIndex({type: "persistent", fields: ["x"], unique: true});
+      const B = fillCollection("B", singleAttributeGenerator(5, "x", x => 2 * x));
+      B.ensureIndex({type: "persistent", fields: ["x"], unique: true});
+
+      const result = runAndCheckQuery(`
+        FOR doc1 IN A
+          SORT doc1.x
+          FOR doc2 IN B
+              FILTER doc1.x == doc2.x
+              RETURN [doc1, doc2]
+      `);
+
+      assertEqual(result.length, 5);
+      for (const [a, b] of result) {
+        assertEqual(a.x, b.x);
+      }
+    },
+
+    testEveryOtherMatchUniqueMedium: function () {
+      const A = fillCollection("A", singleAttributeGenerator(1000, "x", x => x));
+      A.ensureIndex({type: "persistent", fields: ["x"], unique: true});
+      const B = fillCollection("B", singleAttributeGenerator(500, "x", x => 2 * x));
+      B.ensureIndex({type: "persistent", fields: ["x"], unique: true});
+
+      const result = runAndCheckQuery(`
+        FOR doc1 IN A
+          SORT doc1.x
+          FOR doc2 IN B
+              FILTER doc1.x == doc2.x
+              RETURN [doc1, doc2]
+      `);
+
+      assertEqual(result.length, 500);
+      for (const [a, b] of result) {
+        assertEqual(a.x, b.x);
+      }
+    },
+
+    testEveryOtherMatchUniqueLarge: function () {
+      const A = fillCollection("A", singleAttributeGenerator(10000, "x", x => x));
+      A.ensureIndex({type: "persistent", fields: ["x"], unique: true});
+      const B = fillCollection("B", singleAttributeGenerator(5000, "x", x => 2 * x));
+      B.ensureIndex({type: "persistent", fields: ["x"], unique: true});
+
+      const result = runAndCheckQuery(`
+        FOR doc1 IN A
+          SORT doc1.x
+          FOR doc2 IN B
+              FILTER doc1.x == doc2.x
+              RETURN [doc1, doc2]
+      `);
+
+      assertEqual(result.length, 5000);
+      for (const [a, b] of result) {
+        assertEqual(a.x, b.x);
+      }
+    },
```

Comment:
Maybe we want to move these tests into a separate file, like `aql-index-join-unique.js`?